### PR TITLE
SSD16xx implement read support

### PR DESCRIPTION
**Pull request by andysan**
This pull request adds support for basic read operations on the SSD16xx display controller. It consists of two commits:

- Support for read commands and an implementation of the `display_read` API.
- The addition of a device-specific API that lets the caller read out the contents of the second ("red") RAM buffer.

Read operations require the SPI bus to be in half duplex mode. To enable read operations, set the device mode to half duplex by adding the following to the device's DTB node:

      duplex = <SPI_HALF_DUPLEX>;

If read requests will fail with -ENOTSUP if the device isn't in half duplex mode.

---

Tested on a couple of different SSD168x-basde displays using the SPI bit banging driver.

Update: Pushed a new version which moves the device-specific include from the first commit (incorrect) to the second commit. This fixes potential future bisection issues.

Update: Rebased on latest main.

### CODE REVIEW

**Review by jfischer-no — DISMISSED** [review]
:+1:

File: drivers/display/ssd16xx.c
Comment (andysan) [reply drivers/display/ssd16xx.c:373]: Reordered to make sure all consts are first in the declaration block. I kept it under 80 characters though.

File: include/zephyr/display/ssd16xx.h
Comment (jfischer-no) [inline include/zephyr/display/ssd16xx.h:43]: Do you think it would make sense to extend `struct display_buffer_descriptor `so that `int display_read()` could be used instead of driver specific? Possibly also for writing, and I think also about your RFC in discussions.

**Review by jfischer-no — DISMISSED** [review]

**Comment by andysan** [comment]
Ping. Can I get another review on these changes so we can merge them?

**Review by mbolivar-nordic — APPROVED** [review]
@jfischer-no I am approving only after skimming the changes quickly as all seems OK from a high level. Please approve if you believe the details are correct. I do not know this part.

**Review by jfischer-no — APPROVED** [review]

**Comment by andysan** [comment]
Can we make sure that https://github.com/zephyrproject-rtos/zephyr/pull/47712 is merged before this one? These two PRe will be conflicting and I think it would be much less pain for everyone involve if we resolve the conflicts on this smaller PR.

**Pull request merged**
